lib/saml/responses/base.rb

```diff
@@ -62,7 +62,8 @@ def map_message_to_error(error_message)
       end
 
       def authn_context_text
-        REXML::XPath.first(decrypted_document, '//saml:AuthnContextClassRef')&.text
+        response_doc = assertion_encrypted? ? decrypted_document : document
+        REXML::XPath.first(response_doc, '//saml:AuthnContextClassRef')&.text
       end
 
       def authn_context
```

Comment:
is `decrypted_document` truthy or falsy in the case we ran into with testing? Because if the latter, then won't  this never even get to calling `authn_context_text`?

If this feels to tricky to unwind, we can also consider just asking Prashant to enable encrypted assertions. It might make development a little tougher,  but. we have the ability to attach a. debugger and get the decrypted assertion if we need it.  Your call on that one.

Reply:
@patrickvinograd `decrypted_document` is nil in the case we ran into. In `/lib/saml/user.rb`, we call `authn_context_text` directly, which is why we got the 500—it will always get called when the response is parsed.

https://github.com/department-of-veterans-affairs/vets-api/blob/35b2d98c278c65132f4c9c9654b40e3bf912bae2/lib/saml/user.rb#L76

A little confusing but in the user object (`lib/saml/user.rb`), `authn_context` calls `saml_response.authn_context_text`, and in the _response object_ (`lib/saml/responses/base.rb`)  calling authn_context will return `SAML::User::UNKNOWN_AUTHN_CONTEXT` if the assertion is not encrypted.

I left the `authn_context` method on the SAML response class untouched because there are existing session controller specs that explicitly check for `SAML::User::UNKNOWN_AUTHN_CONTEXT` if decrypted_document is falsey, whereas for the purposes of our test and the use case in `lib/saml/user.rb`, it is fine for `authn_context` to return `nil`

Reply:
Yeesh, yeah, I did not follow that from the git diff :) 👍